ZenPacks/zenoss/ZenPackLib/lib/base/ZenPack.py

```diff
@@ -285,16 +285,33 @@ def template_changed(self, app, existing, new_spec):
         return diff
 
     def create_device_class(self, app, dcspec):
-        ''''''
+        """Return existing or created device class"""
+
+        def handle_plugins(d_class, value, overwrite):
+            """ Treat zCollectorPlugins as special case
+                and append new plugins 
+            """
+            if overwrite:
+                d_class.setZenProperty('zCollectorPlugins', value)
+            else:
+                plugins = [p for p in d_class.zCollectorPlugins]
+                plugins.extend([v for v in value if v not in plugins])
+                d_class.setZenProperty('zCollectorPlugins', plugins)
+
+        exists = False
         try:
             dcObject = app.dmd.Devices.getOrganizer(dcspec.path)
+            exists = True
         except KeyError:
             self.LOG.info('Creating DeviceClass {}'.format(dcspec.path))
             dcObject = app.dmd.Devices.createOrganizer(dcspec.path)
 
         for zprop, value in dcspec.zProperties.iteritems():
             if dcObject.getPropertyType(zprop) is None:
                 self.LOG.error("Unable to set zProperty {} on {} (undefined zProperty)".format(zprop, dcspec.path))
+            # modeler plugins are a special case for preexisting device class
```

Comment:
If you think this will come up in other contexts, then perhaps I should make this parameter an attribute of ZPropertySpec instead of DeviceClassSpec?